src/command_modules/azure-cli-interactive/azclishell/_dump_commands.py

```diff
@@ -18,122 +18,116 @@
-            com_descrip['help'] = command_description
-            com_descrip['examples'] = ""
-
-            for key in CMD_TABLE[cmd].arguments:
-                required = ""
-                help_desc = ""
-                if CMD_TABLE[cmd].arguments[key].type.settings.get('required'):
-                    required = "[REQUIRED]"
-                if CMD_TABLE[cmd].arguments[key].type.settings.get('help'):
-                    help_desc = CMD_TABLE[cmd].arguments[key].type.settings.get('help')
-
-                name_options = []
-                for name in CMD_TABLE[cmd].arguments[key].options_list:
-                    name_options.append(name)
-
-                options = {
-                    'name': name_options,
-                    'required': required,
-                    'help': help_desc
-                }
-                param_descrip[CMD_TABLE[cmd].arguments[key].options_list[0]] = options
-
-            com_descrip['parameters'] = param_descrip
-            data[cmd] = com_descrip
-        except (ImportError, ValueError):
-            pass
-
-    for cmd in helps:
-        diction_help = yaml.load(helps[cmd])
-        if "short-summary" in diction_help:
-            if cmd in data:
-                data[cmd]['help'] = diction_help["short-summary"]
-            else:
-                data[cmd] = {
-                    'help': diction_help["short-summary"],
-                    'parameters': {}
-                }
-            if callable(data[cmd]['help']):
-                data[cmd]['help'] = data[cmd]['help']()
-
-        if cmd not in data:
-            print("Command: {} not in Command Table".format(cmd))
-            continue
-
-        if "parameters" in diction_help:
-            for param in diction_help["parameters"]:
-                if param["name"].split()[0] not in data[cmd]['parameters']:
+
+class DumpTable(object):
+    def __init__(self):
+        self.command_table = None
+
+    def install_modules(self):
+        installed_command_modules = []
+        for cmd in self.command_table:
+            try:
+                self.command_table[cmd].load_arguments()
+            except (ImportError, ValueError):
+                pass
+            mods_ns_pkg = import_module('azure.cli.command_modules')
+        for _, modname, _ in pkgutil.iter_modules(mods_ns_pkg.__path__):
+            if modname not in BLACKLISTED_MODS:
+                installed_command_modules.append(modname)
+
+        for mod in installed_command_modules:
+            try:
+                mod = import_module('azure.cli.command_modules.' + mod)
+                mod.load_params(mod)
+                mod.load_commands()
+
+            except Exception:  # pylint: disable=broad-except
+                print("Error loading: {}".format(mod))
+        _update_command_definitions(self.command_table)
+
+    def dump_command_table(self):
+        """ dumps the command table """
+        APPLICATION.initialize(Configuration())
```

Comment:
`APPLICATION` here is a global variable. Initializing it won't introduce unexpected behaviour?